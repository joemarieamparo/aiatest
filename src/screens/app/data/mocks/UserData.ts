import R from "../../../../assets";
import { Article } from "../model/Article";
import { Liker } from "../model/Liker";
import { Profile } from "../model/Profile";
import { UserProfile } from "../model/UserProfile";

const loremIpsum = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'
const description =`${loremIpsum}\n${loremIpsum}\n${loremIpsum}\n${loremIpsum}\n${loremIpsum}\n${loremIpsum}\n${loremIpsum}\n${loremIpsum}\n${loremIpsum}\n${loremIpsum}`
const getProfile = (id: number, name: string, avatar: any, title: string, isCurrentUser?: boolean): Profile => {
    return {
        id,
        name,
        avatar,
        title,
        isCurrentUser
    }
}

const userProfile = getProfile(1, 'John Doe', R.images.userAvatar, 'I am the user of the application', true)
const user1Profile = getProfile(2, 'Mary Jean', R.images.otherUser1, 'I am another user of the application')
const user2Profile = getProfile(3, 'Mary Ann', R.images.otherUser2, 'I am another user of the application')

const article = (id: number, titleNumber: number, thumbnail: any, author: string, likers?: Liker[]): Article => {
    return {
        id,
        title: `Article #${titleNumber}`,
        description: description,
        likers: likers ?? [],
        date: 'May 5, 2022',
        thumbnail, 
        author
    }
}

const articles = (
    thumbnails: any[], 
    author: string, 
    articleIds: number[],
    likers?: Liker[]
): Article[] => {
    return [
        article(articleIds[0], 1, thumbnails[0], author, likers), 
        article(articleIds[1], 2, thumbnails[1], author, likers)
    ]
} 

const userProfileData: UserProfile = {
    profile: userProfile,
    followers: [user1Profile, user2Profile],
    articles: articles(
        [R.images.thumbnail_3, R.images.thumbnail_4], 
        userProfile.name, 
        [10, 20],
        [{id: user1Profile.id, articleId: 10}]
    )
}

const user1ProfileData: UserProfile = {
    profile: user1Profile,
    followers: [userProfile, user2Profile],
    articles: articles(
        [R.images.thumbnail_3, R.images.thumbnail_4], 
        user1Profile.name, 
        [30, 40],
        [{id: user2Profile.id, articleId: 40}])
}

const user2ProfileData: UserProfile = {
    profile: user2Profile,
    followers: [userProfile, user1Profile],
    articles: articles(
        [R.images.thumbnail_5, R.images.thumbnail_6], 
        user2Profile.name, 
        [50, 60],
        [{id: user1Profile.id, articleId: 50}])
}

export const getMockProfile = (name: string): UserProfile => {
    switch(name) {
        case 'Mary Jean': return user1ProfileData
        case 'Mary Ann': return user2ProfileData
        default: return userProfileData
    }
}

