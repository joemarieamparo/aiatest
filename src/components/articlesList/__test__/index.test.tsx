import React from 'react'
import {render} from '@testing-library/react-native'
import ArticleList from '../index'
import { getMockProfile } from '../../../screens/app/data/mocks/UserData'

const mockNavigate = jest.fn()
const goBack = jest.fn()
const createTestProps = (propsData: Object) => ({
    navigation: {
      navigate: mockNavigate,
      state: jest.fn(),
      goBack
    },
    ...propsData
  })
describe('Screen: ArticleList', () => {
    it('ArticleList: renders correctly', () => {
        const props: any = createTestProps({
          articles: getMockProfile('John Doe'),
          onItemClick: jest.fn(),
          header: 'Test Header'
        })
        const {toJSON} = render(<ArticleList {...props} />)
        expect(toJSON()).toMatchSnapshot();
    })
})