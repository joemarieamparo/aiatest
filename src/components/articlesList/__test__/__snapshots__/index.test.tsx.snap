// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen: ArticleList ArticleList: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "bold",
        "margin": 8,
      }
    }
    testID="text_text_id"
  >
    Test Header
  </Text>
  <RCTScrollView
    data={
      Object {
        "articles": Array [
          Object {
            "date": "May 5, 2022",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
            "likers": Array [
              Object {
                "avatar": Object {
                  "testUri": "../../../src/assets/images/mary_jean.jpeg",
                },
                "isCurrentUser": undefined,
                "name": "Mary Jean",
                "title": "I am another user of the application",
              },
            ],
            "thumbnail": Object {
              "testUri": "../../../src/assets/images/thumbnail_3.jpeg",
            },
            "title": "Article #1",
          },
          Object {
            "date": "May 5, 2022",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
            "likers": Array [
              Object {
                "avatar": Object {
                  "testUri": "../../../src/assets/images/mary_jean.jpeg",
                },
                "isCurrentUser": undefined,
                "name": "Mary Jean",
                "title": "I am another user of the application",
              },
            ],
            "thumbnail": Object {
              "testUri": "../../../src/assets/images/thumbnail_4.jpeg",
            },
            "title": "Article #2",
          },
        ],
        "followers": Array [
          Object {
            "avatar": Object {
              "testUri": "../../../src/assets/images/mary_jean.jpeg",
            },
            "isCurrentUser": undefined,
            "name": "Mary Jean",
            "title": "I am another user of the application",
          },
          Object {
            "avatar": Object {
              "testUri": "../../../src/assets/images/mary_ann.jpeg",
            },
            "isCurrentUser": undefined,
            "name": "Mary Ann",
            "title": "I am another user of the application",
          },
        ],
        "profile": Object {
          "avatar": Object {
            "testUri": "../../../src/assets/images/john_doe.jpeg",
          },
          "isCurrentUser": true,
          "name": "John Doe",
          "title": "I am the user of the application",
        },
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
